### PR TITLE
Feature/specimen endpoint

### PR DESCRIPTION
**Pull request by caaespin**
Adding specimen endpoint to the webservice

We can do refactoring/ cosmetic stuff after the demo

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -113,15 +113,88 @@ def get_data(file_id=None):
         response = dict(error=bae.message)
         response.status_code = 400
         return response
-    except Exception as e:
-        logger.error("Malformed filters parameter: {}".format(e))
-        return "Malformed filters parameter"
+    # except Exception as e:
```
Comment (hannes-ucsc) [inline app.py:116]: Do you really want this commented out?

File: chalicelib/responseobjects/hca_response_v5.py [hunk 2]
```diff
@@ -350,8 +353,8 @@ def make_specimens(self, entry):
                     translated_specimen["totalCells"] = sum(translated_specimen["totalCells"])
                     specimens[specimen_id] = translated_specimen
                 else:
-                    merged_specimen = self._merge(specimens[specimen_id], translated_specimen, "biomaterial_id")
-                    merged_specimen["total_cells"] = sum(merged_specimen["total_cells"])
+                    merged_specimen = self._merge(specimens[specimen_id], translated_specimen, "id")
+                    merged_specimen["totalCells"] = sum(merged_specimen["totalCells"])
```
Comment (hannes-ucsc) [inline chalicelib/responseobjects/hca_response_v5.py:357]: For future reference: I'd prefer snake_case for fields everywhere.

File: chalicelib/responseobjects/hca_response_v5.py [hunk 3]
```diff
@@ -298,13 +301,13 @@ def make_projects(self, entry):
             project_id = project["project"]
             translated_project = {
                 "shortname": project_id,
-                "laboratory": list(set(project.get("laboratory", None)))
+                "laboratory": list(set(project.get("laboratory")) if project.get("laboratory") else [])
```
Comment (hannes-ucsc) [inline chalicelib/responseobjects/hca_response_v5.py:304]: `"laboratory": list(set(project.get("laboratory", [])))`

File: chalicelib/responseobjects/hca_response_v5.py [hunk 4]
```diff
@@ -273,6 +273,9 @@ def make_processes(self, entry):
         processes = {}
         for bundle in entry["bundles"]:
             for process in bundle["contents"]["processes"]:
+                # HACK: need to check on the indexer why raw protocols are being added
```
Comment (hannes-ucsc) [inline chalicelib/responseobjects/hca_response_v5.py:276]: How will you remember to do this?

Anyhow, this is exactly why we need to merge the two repos. My eyes are burning looking at all this tight coupling between separate projects.

File: config/chalice/config.json [hunk 5]
```diff
@@ -1,16 +1,19 @@
...
       "manage_iam_role":false,
-      "iam_role_arn":"arn:aws:iam::<AWS_ACCOUNT_ID>:role/<DASHBOARD_LAMBDA_APPLICATION_NAME>-dev",
-      "environment_variables": {
-         "ES_ENDPOINT":<ELASTICSEARCH_ENDPOINT>,
-         "ES_INDEX":"<INDEX_TO_USE>",
-         "DASHBOARD_NAME":"<DASHBOARD_LAMBDA_APPLICATION_NAME>"
-      }
+      "iam_role_arn":"arn:aws:iam::861229788715:role/browser-api-dev",
+     	 "environment_variables": {
+         "ES_DOMAIN":"search-azul-index-dev-scdexoaebx3aafy7cei4az3ivy.us-east-1.es.amazonaws.com",
```
Comment (hannes-ucsc) [inline config/chalice/config.json:10]: Make sure this is the right domain.

**Review by hannes-ucsc — APPROVED** [review]

File: chalicelib/responseobjects/hca_response_v5.py (hunk 3 above)
Comment (caaespin) [reply chalicelib/responseobjects/hca_response_v5.py:304]: The problem is that project["laboratory"] can actually be None.

File: config/chalice/config.json (hunk 5 above)
Comment (caaespin) [reply config/chalice/config.json:10]: We will use this instance for now.

**Pull request merged**